app/services/zendesk_follow_up_docs_service.rb

```diff
@@ -12,10 +11,9 @@ def send_requested_docs
     new_requested_docs = @intake.documents.where(document_type: "Requested", zendesk_ticket_id: nil)
     file_list = new_requested_docs.map do |document|
       @document_blob = document.upload
-      tmpfile = Tempfile.open(["ZendeskFollowUpDocsService", blob.filename.extension_with_delimiter], Dir.tmpdir)
-      download_blob_to(tmpfile)
-
-      { file: tmpfile, filename: document.upload.filename.to_s }
+      blob.open(tmpdir: Dir.tmpdir) do |file|
+        {file: file, filename: document.upload.filename.to_s}
```

Comment:
@tdooner it sounds like refactoring the doc handling inside of the intake service would be a necessary step to upgrading, correct?

Reply:
That's my impression. I believe if we merged this, we would get errors when we tried to upload docs to Zendesk because the downloaded S3 tempfiles would be deleted before the `attach_multiple_files_to_ticket` method is called. That said... I think this refactor is not too large (could be an hour or two of pairing).

Reply:
Tom's suggested fix makes sense to me. I wasn't sure if we could get away with the simpler approach I wrote, and I couldn't test it, but the recursive method will definitely work. Happy to pair on it whenever we have time.